idx/memory/memory.go

```diff
@@ -59,6 +61,7 @@ func ConfigSetup() {
 	memoryIdx.BoolVar(&TagSupport, "tag-support", false, "enables/disables querying based on tags")
 	memoryIdx.IntVar(&TagQueryWorkers, "tag-query-workers", 50, "number of workers to spin up to evaluate tag queries")
 	memoryIdx.IntVar(&matchCacheSize, "match-cache-size", 1000, "size of regular expression cache in tag query evaluation")
+	memoryIdx.DurationVar(&maxPruneLockTime, "max-prune-lock-time", time.Millisecond*100, "Maximum duration each second a prune job can lock the index.")
```

Comment:
we have no proper way to validate duration flags. (see #944)
thus my recommendation is:
* use a StringVar
* create a ConfigProcess() function where we can explicitly parse and fail if needed
* call ConfigProcess() from the `Validate remaining settings` section in metrictank.go

Reply:
I really dont like this model of config mgmt.  MemoryIdx already has an `Init() error` function, can we not just validated the config settings there?